.github/workflows/docker-tests.yml

```diff
@@ -0,0 +1,189 @@
+      - 'nifi-toolkit/nifi-toolkit-assembly/**'
+      - 'nifi-toolkit/nifi-toolkit-cli/**'
+      - 'nifi-toolkit/nifi-toolkit-encrypt-config/**'
+      - 'nifi-toolkit/nifi-toolkit-tls/**'
+      - 'minifi/minifi-assembly/**'
+      - 'minifi/minifi-docker/**'
+      - 'minifi/minifi-c2/minifi-c2-assembly/**'
+      - 'minifi/minifi-c2/minifi-c2-docker/**'
+  pull_request:
+    paths:
+      - '.github/workflows/docker-tests.yml'
+      - 'nifi-assembly/**'
+      - 'nifi-docker/**'
+      - 'nifi-registry/nifi-registry-assembly/**'
+      - 'nifi-registry/nifi-registry-docker-maven/**'
+      - 'nifi-toolkit/nifi-toolkit-assembly/**'
+      - 'nifi-toolkit/nifi-toolkit-cli/**'
+      - 'nifi-toolkit/nifi-toolkit-encrypt-config/**'
+      - 'nifi-toolkit/nifi-toolkit-tls/**'
+      - 'minifi/minifi-assembly/**'
+      - 'minifi/minifi-docker/**'
+      - 'minifi/minifi-c2/minifi-c2-assembly/**'
+      - 'minifi/minifi-c2/minifi-c2-docker/**'
+
+env:
+  DEFAULT_MAVEN_OPTS: >-
+    -Xmx4g
+    -XX:ReservedCodeCacheSize=1g
+    -XX:+UseG1GC
+    -Dorg.slf4j.simpleLogger.defaultLogLevel=WARN
+    -Daether.connector.http.retryHandler.count=5
+    -Daether.connector.http.connectionMaxTtl=30
+  MAVEN_COMMAND: >-
+    ./mvnw
+    -V
+    -nsu
+    -ntp
+    -ff
+
+# build assemblies (including binaries) for use in Docker Image builds
+  MAVEN_BUILD_ARGUMENTS: >-
+    -am
+    -D skipTests
+    -T 2.0C
+    install
+  MAVEN_BUILD_PROJECTS: >-
+    -pl nifi-assembly
+    -pl nifi-registry/nifi-registry-assembly
+    -pl nifi-toolkit/nifi-toolkit-assembly
+    -pl minifi/minifi-assembly
+    -pl minifi/minifi-c2/minifi-c2-assembly
+
+  MAVEN_CLEAN_ARGUMENTS: >-
+    -T 2.0C
+    clean
+  MAVEN_CLEAN_PROJECTS: >-
+    -pl -nifi-assembly
+    -pl -nifi-registry/nifi-registry-assembly
+    -pl -nifi-toolkit/nifi-toolkit-assembly
+    -pl -nifi-toolkit/nifi-toolkit-cli
+    -pl -nifi-toolkit/nifi-toolkit-encrypt-config
+    -pl -nifi-toolkit/nifi-toolkit-tls
+    -pl -minifi/minifi-assembly
+    -pl -minifi/minifi-c2/minifi-c2-assembly
+
+# build and test Docker Images
+  MAVEN_DOCKER_ARGUMENTS: >-
+    verify
+    -P docker
+  MAVEN_DOCKER_PROJECTS: >-
+    -pl minifi/minifi-docker
+    -pl minifi/minifi-c2/minifi-c2-docker
+    -pl nifi-docker/dockermaven
+    -pl nifi-registry/nifi-registry-docker-maven/dockermaven
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  build_and_test:
+    strategy:
+      fail-fast: false
+      matrix:
+        # TODO: macos-latest doesn't come with Docker and Colima seems to have problems with the nifi-docker integration_test scripts - see NIFI-12177/NIFI-12191
+        os: [ ubuntu-latest ]
+        version: [ 21 ]
+    timeout-minutes: 120
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }} Java ${{ matrix.version }}
+    env:
+      # default to Amazon Corretto JDK
+      JAVA_DISTRIBUTION: corretto
```

Comment:
In this case, it seems much simpler to use `Zulu` for all builds. The JDK version is less important for this workflow because the focus is running Docker containers.